File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/domain/entity/LibrarianEntity.java
package ru.elenacherev.librarymanager.domain.entity;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "LIBRARIANS")
@Getter
@NoArgsConstructor
//@RequiredArgsConstructor
@Setter
public class LibrarianEntity {
    @Id
    @GeneratedValue(strategy = IDENTITY) //уникальный ID, генерируется сам в БД
    @Column(name = "LIBRARIAN_ID")
    @NotNull
    @Setter(AccessLevel.NONE)
    private Long librarianId;

    @Version
    @Column(name = "VERSION")
    private int version; //версия, инкрементится при редактировании записи

}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/domain/entity/LibrarianEntity.kt
package ru.elenacherev.librarymanager.domain.entity

import org.hibernate.annotations.ColumnDefault
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.Table
import javax.persistence.Version
import javax.validation.constraints.NotNull

@Entity
@Table(name = "LIBRARIANS")
data class LibrarianEntity(

    @Column(name = "LIBRARIAN_ID", nullable = false, updatable = false)
    @GeneratedValue
    @Id
    @get:NotNull
    var librarianId: UUID? = null,

    //версия, инкрементится при редактировании записи
    @Column(name = "VERSION", nullable = false)
    @Version
    var version: Int = 0
)